ui/js/actions/app.js

```diff
@@ -64,6 +64,17 @@ export function doHistoryBack() {
     if (!history.state) return;
 
     history.back();
+    dispatch({
+      type: types.HISTORY_BACK,
```

Comment:
Is this supposed to be `HISTORY_BACK_COMPLETED`? If not, I don't think `doHistoryBackCompleted` ever happens. If so, I don't think we're guaranteed the render happens while `navigatingBack` is true.

The other relevant code to know here (if you haven't discovered it already) is the popstate event listener in `ui/js/main.js`. This is what history.back() is triggering.